stablehlo/dialect/TypeInference.cpp

```diff
@@ -3523,71 +3524,73 @@ LogicalResult inferWhileOp(std::optional<Location>, ValueRange operand,
 // Verifiers for ops.
 //===----------------------------------------------------------------------===//
 
-LogicalResult verifyAllGatherOp(std::optional<Location> location, Value operand,
-                                int64_t allGatherDim,
+LogicalResult verifyAllGatherOp(std::optional<Location> location,
+                                ValueRange operands, int64_t allGatherDim,
                                 DenseIntElementsAttr replicaGroups,
                                 int64_t channelId, bool useGlobalDeviceIds,
-                                Value result) {
-  auto operandType = cast<RankedTensorType>(operand.getType());
-  auto resultType = cast<RankedTensorType>(result.getType());
-
-  // all_gather_c1
-  if (allGatherDim >= operandType.getRank())
-    return emitOptionalError(location,
-                             "all_gather_dim must be a valid index of operand");
+                                ValueRange results) {
+  for (auto [operand, result] : llvm::zip(operands, results)) {
```

Comment:
Similarly here.

Reply:
`llvm::zip` / `llvm::enumerate` expects declared type must be 'auto' or reference to 'auto'. Thank you.